plugins/inputs/s7comm/s7comm.go

```diff
@@ -0,0 +1,402 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package s7comm
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"hash/maphash"
+	"log" //nolint:depguard // Required for tracing connection issues
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/robinson/gos7"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const maxRequestsPerBatch = 20
+const addressRegexp = `^(?P<area>[A-Z]+)(?P<no>[0-9]+)\.(?P<type>[A-Z]+)(?P<start>[0-9]+)(?:\.(?P<extra>.*))?$`
+
+var (
+	regexAddr = regexp.MustCompile(addressRegexp)
+	// Area mapping taken from https://github.com/robinson/gos7/blob/master/client.go
+	areaMap = map[string]int{
+		"PE": 0x81, // process inputs
+		"PA": 0x82, // process outputs
+		"MK": 0x83, // Merkers
+		"DB": 0x84, // DB
+		"C":  0x1C, // counters
+		"T":  0x1D, // timers
+	}
+	// Word-length mapping taken from https://github.com/robinson/gos7/blob/master/client.go
+	wordLenMap = map[string]int{
+		"X":  0x01, // Bit
+		"B":  0x02, // Byte (8 bit)
+		"C":  0x03, // Char (8 bit)
+		"S":  0x03, // String (8 bit)
+		"W":  0x04, // Word (16 bit)
+		"I":  0x05, // Integer (16 bit)
+		"DW": 0x06, // Double Word (32 bit)
+		"DI": 0x07, // Double integer (32 bit)
+		"R":  0x08, // IEEE 754 real (32 bit)
+		// see https://support.industry.siemens.com/cs/document/36479/date_and_time-format-for-s7-?dti=0&lc=en-DE
+		"DT": 0x0F, // Date and time (7 byte)
+		// "C":  0x1C, // Counter (16 bit)
+		// "T":  0x1D, // Timer (16 bit)
```

Comment:
Why are these commented out? I see them in the areaMap as well. Should these be removed?

Reply:
In S7 protocol, there is the area (which are basically different physical blocks or different memory parts. These are in the "wordlenMap" which basically maps the **data-types** to a identifier. So those are different things. But I will remove them...